notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+    "<i>It seems that the presence of both features 'state' and 'city' isn't necessary, that is why we should delete one of them.<br>I'll choose 'state' feature, because 'city' feature gives us more information (there can be few cities in the state).<br>'city' feature has only 7 more values than 'state' feature, that is why speaking about the model complexity, there should not be much difference.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data = train_data.drop('state', axis=1)\n",
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_copy = train_data.copy()\n",
+    "train_data_copy.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 100))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(17, 2, i+1)\n",
+    "  for city in train_data_copy['city'].unique():\n",
+    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
+    "                                 (train_data_copy['city'] == city)].groupby('date').mean()['sales']\n",
+    "      trend = average_sales.rolling(\n",
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
+    "  plt.title(family_value)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(7,5))\n",
+    "sns.countplot(x=train_data['type'], alpha=0.7, data=train_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 100))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(17, 2, i+1)\n",
+    "  for store_type in train_data_copy['type'].unique():\n",
+    "      #plt.plot(np.arange(len(current_type_and_family_data.index)), current_type_and_family_data['sales'])\n",
+    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
+    "                                 (train_data_copy['type'] == store_type)].groupby('date').mean()['sales']\n",
+    "      trend = average_sales.rolling(\n",
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
+    "  plt.title(family_value)\n",
+    "  plt.legend(train_data_copy['type'].unique())\n",
```

Comment:
Probably, this will fix your issue with the plot legend:
```suggestion
    "  ax = plt.subplot(17, 2, i + 1)\n",
    "  for store_type in train_data_copy['type'].unique():\n",
    "      #plt.plot(np.arange(len(current_type_and_family_data.index)), current_type_and_family_data['sales'])\n",
    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
    "                                 (train_data_copy['type'] == store_type)].groupby('date').mean()['sales']\n",
    "      trend = average_sales.rolling(\n",
    "          window=365,\n",
    "          center=True,\n",
    "          min_periods=183,\n",
    "      ).mean()\n",
    "      trend.plot(ax=ax, linewidth=3, label=store_type)\n",
    "  ax.set_title(family_value)\n",
    "  ax.legend()\n",
```